File: Server/Server-Side/TeamApp/TeamApp.Application/Wrappers/ApiResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TeamApp.Application.Wrappers
{
    public class ApiResponse<T>
    {
        public ApiResponse()
        {
        }
        public ApiResponse(T data, string message = null)
        {
            Succeeded = true;
            Message = message;
            Data = data;
        }
        public ApiResponse(string message)
        {
            Succeeded = false;
            Message = message;
        }
        public bool Succeeded { get; set; }
        public string Message { get; set; }
        public List<string> Errors { get; set; }
        public string ErrorCode { get; set; }
        public T Data { get; set; }
    }
}

File: Client/chat_app/lib/models/api_response.dart
class ApiResponse {
  bool? succeeded = false;
  String? message;
  List<String>? errors;
  String? errorCode;
  dynamic data;

  ApiResponse(
      {this.succeeded, this.message, this.errors, this.errorCode, this.data});
  factory ApiResponse.fromJson(dynamic json) => ApiResponse(
      succeeded: json['succeeded'] as bool?,
      message: json['message'] as String?,
      errors: json['errors'] as List<String>?,
      errorCode: json['errorCode'] as String?,
      data: json['data']);
}
